tests/vmi_iothreads_test.go

```diff
@@ -75,5 +79,148 @@ var _ = Describe("IOThreads", func() {
+			Expect(*disk0.Driver.IOThread).ToNot(Equal(*disk1.Driver.IOThread))
+		})
+
+		table.DescribeTable("[ref_id:2065] should honor auto ioThreadPolicy", func(numCpus int, expectedIOThreads int) {
+			policy := v1.IOThreadsPolicyAuto
+			vmi.Spec.Domain.IOThreadsPolicy = &policy
+
+			dedicated := true
+			vmi.Spec.Domain.Devices.Disks[0].DedicatedIOThread = &dedicated
+
+			tests.AddEphemeralDisk(vmi, "ded2", "virtio", tests.ContainerDiskFor(tests.ContainerDiskCirros))
+			vmi.Spec.Domain.Devices.Disks[1].DedicatedIOThread = &dedicated
+
+			tests.AddEphemeralDisk(vmi, "shr1", "virtio", tests.ContainerDiskFor(tests.ContainerDiskCirros))
+			tests.AddEphemeralDisk(vmi, "shr2", "virtio", tests.ContainerDiskFor(tests.ContainerDiskCirros))
+			tests.AddEphemeralDisk(vmi, "shr3", "virtio", tests.ContainerDiskFor(tests.ContainerDiskCirros))
+			tests.AddEphemeralDisk(vmi, "shr4", "virtio", tests.ContainerDiskFor(tests.ContainerDiskCirros))
+
+			cpuReq := resource.MustParse(fmt.Sprintf("%d", numCpus))
+			vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceCPU] = cpuReq
+
+			By("Creating VMI with 2 dedicated and 4 shared ioThreadPolicies")
+			vmi, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+			Expect(err).ToNot(HaveOccurred())
+
+			tests.WaitForSuccessfulVMIStart(vmi)
+
+			getOptions := metav1.GetOptions{}
+			var newVMI *v1.VirtualMachineInstance
+
+			By("Fetching the VMI from the cluster")
+			newVMI, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(vmi.Name, &getOptions)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Fetching the domain XML from the running pod")
+			domain, err := tests.GetRunningVirtualMachineInstanceDomainXML(virtClient, vmi)
+			Expect(err).ToNot(HaveOccurred())
+			domSpec := &api.DomainSpec{}
+			Expect(xml.Unmarshal([]byte(domain), domSpec)).To(Succeed())
+
+			By("Verifying the total number of ioThreads")
+			Expect(int(domSpec.IOThreads.IOThreads)).To(Equal(expectedIOThreads))
+
+			By("Ensuring there are the expected number of disks")
+			Expect(len(newVMI.Spec.Domain.Devices.Disks)).To(Equal(len(vmi.Spec.Domain.Devices.Disks)))
+
+			By("Verifying the ioThread mapping for disks")
+			disk0, err := getDiskByName(domSpec, "disk0")
+			Expect(err).ToNot(HaveOccurred())
+			ded2, err := getDiskByName(domSpec, "ded2")
+			Expect(err).ToNot(HaveOccurred())
+			shr1, err := getDiskByName(domSpec, "shr1")
+			Expect(err).ToNot(HaveOccurred())
+			shr2, err := getDiskByName(domSpec, "shr2")
+			Expect(err).ToNot(HaveOccurred())
+			shr3, err := getDiskByName(domSpec, "shr2")
+			Expect(err).ToNot(HaveOccurred())
+			shr4, err := getDiskByName(domSpec, "shr2")
+			Expect(err).ToNot(HaveOccurred())
+
+			// the ioThreads for disks sh1 through shr4 will vary based on how many CPUs there are
+			// but we already verified the total number of threads, so we know they're spread out
+			// across the proper threadId pool.
+
+			By("Ensuring disk0 has a unique threadId")
```

Comment:
How do you think, maybe better to dump xml, unmarshall it, and parse it instead of use this function?

Reply:
@sHaggYcaT It's honestly unclear what you're suggesting. lines 164 and 167 of this test are dumping and unmarshalling the domain XML:

```
By("Fetching the domain XML from the running pod")
domain, err := tests.GetRunningVirtualMachineInstanceDomainXML(virtClient, vmi)
Expect(err).ToNot(HaveOccurred())
domSpec := &api.DomainSpec{}
Expect(xml.Unmarshal([]byte(domain), domSpec)).To(Succeed())
```

So I guess the part that's unclear to me is when you suggest "and parse it". What exactly are you picturing?

Reply:
Yeah, you're right @stu-gott